server/routes/user/index.ts

```diff
@@ -0,0 +1,358 @@
+import logger from '../../logger';
+import gravatarUrl from 'gravatar-url';
+import { UserType } from '../../constants/user';
+import { isAuthenticated } from '../../middleware/auth';
+import { UserRequestsResponse } from '../../interfaces/api/userInterfaces';
+import userSettingsRoutes from './usersettings';
+
+const router = Router();
+
+router.get('/', async (req, res) => {
+  let query = getRepository(User).createQueryBuilder('user');
+
+  switch (req.query.sort) {
+    case 'updated':
+      query = query.orderBy('user.updatedAt', 'DESC');
+      break;
+    case 'displayname':
+      query = query.orderBy(
+        '(CASE WHEN user.username IS NULL THEN user.plexUsername ELSE user.username END)',
+        'ASC'
+      );
+      break;
+    case 'requests':
+      query = query
+        .addSelect((subQuery) => {
+          return subQuery
+            .select('COUNT(request.id)', 'requestCount')
+            .from(MediaRequest, 'request')
+            .where('request.requestedBy.id = user.id');
+        }, 'requestCount')
+        .orderBy('requestCount', 'DESC');
+      break;
+    default:
+      query = query.orderBy('user.id', 'ASC');
+      break;
+  }
+
+  const users = await query.getMany();
+
+  return res
+    .status(200)
+    .json(
+      User.filterMany(users, req.user?.hasPermission(Permission.MANAGE_USERS))
+    );
+});
+
+router.post(
+  '/',
+  isAuthenticated(Permission.MANAGE_USERS),
+  async (req, res, next) => {
+    try {
+      const settings = getSettings();
+
+      const body = req.body;
+      const userRepository = getRepository(User);
+
+      const passedExplicitPassword = body.password && body.password.length > 0;
+      const avatar = gravatarUrl(body.email, { default: 'mm', size: 200 });
+
+      if (!passedExplicitPassword && !settings.notifications.agents.email) {
+        throw new Error('Email notifications must be enabled');
+      }
+
+      const user = new User({
+        avatar: body.avatar ?? avatar,
+        username: body.username ?? body.email,
+        email: body.email,
+        password: body.password,
+        permissions: settings.main.defaultPermissions,
+        plexToken: '',
+        userType: UserType.LOCAL,
+      });
+
+      if (passedExplicitPassword) {
+        await user?.setPassword(body.password);
+      } else {
+        await user?.generatePassword();
+      }
+
+      await userRepository.save(user);
+      return res.status(201).json(user.filter());
+    } catch (e) {
+      next({ status: 500, message: e.message });
+    }
+  }
+);
+
+router.get<{ id: string }>('/:id', async (req, res, next) => {
+  try {
+    const userRepository = getRepository(User);
+
+    const user = await userRepository.findOneOrFail({
+      where: { id: Number(req.params.id) },
+    });
+
+    return res
+      .status(200)
+      .json(user.filter(req.user?.hasPermission(Permission.MANAGE_USERS)));
+  } catch (e) {
+    next({ status: 404, message: 'User not found' });
```

Comment:
```suggestion
    next({ status: 404, message: 'User not found.' });
```

I noticed that some error messages end with punctuation while others do not.  I know this isn't new code, but this is a good opportunity to fix that in this file at least 😄

Reply:
No one sees this really but I will change it for your sake :D